src/OrchardCore.Modules/Orchard.Liquid/Drops/QueriesDrop.cs

```diff
@@ -32,6 +32,11 @@ public override object BeforeMethod(string method)
                 return ((IEnumerable<JObject>)result).Select(x => new JTokenDrop(x)).ToArray();
             }
 
+            if (result is IEnumerable<object>)
```

Comment:
Isn't the issue that the SqlQuerySource should return something that implements `IEnumerable<JObject>` instead?